### PR TITLE
Install iptables in docker containers

### PR DESCRIPTION
**Pull request by rmstar**
Update dockerfiles to install install iptables. Iptables is required for some network state transition tests that I'm working on.

### CODE REVIEW

**Comment by jtattermusch** [comment]
@rmstar  please provide more detail about what you want to test and how. Also, it's unlikely you'll need iptables in all the docker images out there.  Can you add iptables where you really need them (e.g. just to the C++ docker image).
Btw, these tests won't work with bazel RBE, are you planning to just disable them in bazel?

**Comment by rmstar** [comment]
@jtattermusch I want to test the behavior of gRPC when the network connection is flaky. This is especially relevant to mobile, where network blips can expose issues in core.  See #17226 for example. 
I have a PR open for my test: #17903.

I just need iptables in the container that runs C++ tests. So I need to update `templates/tools/dockerfile/cxx_deps.include`?
Also, can you please explain why the tests won't work with bazel RBE? Thanks!

**Comment by rmstar** [comment]
objc_macos_opt_native failure looks like #17814
```
Testing failed:
	No such file or directory: '/Volumes/BuildData/tmpfs/src/github/grpc/workspace_objc_macos_opt_native/src/core/lib/gpr/log_posix.cc'
	No input files
** TEST FAILED **
```

grpc_python_bazel_test failed with error: 
```
FAIL: //src/python/grpcio_tests/tests/unit:_abort_test (see /root/.cache/bazel/_bazel_root/d2dc70c3d9da3fab488ba0dcbbd35051/execroot/com_github_grpc_grpc/bazel-out/k8-fastbuild/testlogs/src/python/grpcio_tests/tests/unit/_abort_test/test.log)
INFO: From Testing //src/python/grpcio_tests/tests/unit:_abort_test:
==================== Test output for //src/python/grpcio_tests/tests/unit:_abort_test:
test_abort (__main__.AbortTest) ... ok
test_abort_does_not_leak_local_vars (__main__.AbortTest) ... FAIL
test_abort_with_status (__main__.AbortTest) ... ok
test_invalid_code (__main__.AbortTest) ... ok
======================================================================
FAIL: test_abort_does_not_leak_local_vars (__main__.AbortTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/root/.cache/bazel/_bazel_root/d2dc70c3d9da3fab488ba0dcbbd35051/execroot/com_github_grpc_grpc/bazel-out/k8-fastbuild/bin/src/python/grpcio_tests/tests/unit/_abort_test.runfiles/com_github_grpc_grpc/src/python/grpcio_tests/tests/unit/_abort_test.py", line 130, in test_abort_does_not_leak_local_vars
    self.assertIsNone(weak_ref())
AssertionError: <__main__._Object object at 0x7fc0eee5bd50> is not None
----------------------------------------------------------------------
Ran 4 tests in 0.116s
FAILED (failures=1)
``` 
I don't see an open issue for this, will open one.

TSAN C++ failure is #17335.

**Comment by rmstar** [comment]
@jtattermusch I have restricted my changes to templates/tools/dockerfile/test/cxx_jessie_x64/Dockerfile.template, so only the container used for running c++ tests will be updated.

File: templates/tools/dockerfile/test/cxx_jessie_x64/Dockerfile.template [hunk 1]
```diff
@@ -25,7 +25,7 @@
   <%include file="../../libuv_install.include"/>
 
   # Install gcc-4.8 and other relevant items
-  RUN apt-get update && apt-get -y install gcc-4.8 gcc-4.8-multilib g++-4.8 g++-4.8-multilib && apt-get clean
+  RUN apt-get update && apt-get -y install gcc-4.8 gcc-4.8-multilib g++-4.8 g++-4.8-multilib iptables && apt-get clean
```
Comment (jtattermusch) [inline templates/tools/dockerfile/test/cxx_jessie_x64/Dockerfile.template]: let's add iptables on a separate line and add an explanatory comment.

`RUN apt-get update && apt-get -y install iptables && apt-get clean`

Comment (jtattermusch) [inline templates/tools/dockerfile/test/cxx_jessie_x64/Dockerfile.template]: Also, there's are more "cxx" docker files so I'm not sure if omitting iptables from them will break things - you need to check their usage.

https://github.com/grpc/grpc/search?q=cxx_deps.include&unscoped_q=cxx_deps.include

**Comment by jtattermusch** [comment]
> @jtattermusch I want to test the behavior of gRPC when the network connection is flaky. This is especially relevant to mobile, where network blips can expose issues in core. See #17226 for example.
> I have a PR open for my test: #17903.
> 
> I just need iptables in the container that runs C++ tests. So I need to update `templates/tools/dockerfile/cxx_deps.include`?
> Also, can you please explain why the tests won't work with bazel RBE? Thanks!

Bazel RBE is a managed test execution framework and I think the capabilities are somewhat restricted.
That said, the tests on Bazel RBE do run exclusively (1VM executes just one test at a time), but I don't think they run as root (which is a good thing in general) and it might not be possible to run set all the settings the way you need to. Also, setting a custom image for Bazel RBE (one that has iptables) is something we haven't tried.

**Comment by rmstar** [comment]
This change is not required because we're installing iptables as part of the test setup in #17903. I'll close this PR.

**Pull request closed without merging**